apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ValueConversion.cs

```diff
@@ -252,9 +248,37 @@ private static object ConvertToClrTypeImpl(Value wireValue, V1.Type spannerType,
                     case Value.KindOneofCase.NullValue:
                         return default(bool);
                     case Value.KindOneofCase.StringValue:
+                        if (spannerType.Code == TypeCode.Int64)
+                        {
+                            return Convert.ToBoolean(Convert.ToInt64(wireValue.StringValue));
+                        }
                         return Convert.ToBoolean(wireValue.StringValue);
                     case Value.KindOneofCase.BoolValue:
                         return wireValue.BoolValue;
+                    case Value.KindOneofCase.NumberValue:
+                        return Convert.ToBoolean(wireValue.NumberValue);
+                    default:
+                        throw new InvalidOperationException(
+                            $"Invalid Type conversion from {wireValue.KindCase} to {targetClrType.FullName}");
+                }
+            }
+
+            if (targetClrType == typeof(char))
+            {
+                switch (wireValue.KindCase)
+                {
+                    case Value.KindOneofCase.BoolValue:
+                        return Convert.ToChar(wireValue.BoolValue);
+                    case Value.KindOneofCase.NumberValue:
+                        return Convert.ToChar(wireValue.NumberValue);
+                    case Value.KindOneofCase.NullValue:
+                        return default(char);
+                    case Value.KindOneofCase.StringValue:
+                        if (spannerType.Code == TypeCode.Int64)
```

Comment:
hopefully this makes more sense now.  int64 spanner types are encoded over the wire as strings.